<!-- Runs obstacle detector and tracker with two Hokuyo laser scanners.
     Uses mocap_optitrack, laser_scan_matcher or static_tf for world to base transformation. -->
<launch>
  <node pkg="hokuyo_node" type="hokuyo_node" name="front_hokuyo" output="screen" respawn="true">
    <param name="port"           value="/dev/sensors/hokuyo_H1320250"/>
    <param name="frame_id"       value="front_scanner"/>

    <param name="intensity"      value="false"/>
    <param name="calibrate_time" value="true"/>

    <param name="min_ang"        value="-2.0862"/>
    <param name="max_ang"        value="2.0923"/>
    <param name="skip"           value="0"/>
    <param name="time_offset"    value="0.0"/>
    <param name="cluster"        value="1"/>

    <remap from="scan" to="front_scan"/>
  </node>

  <node pkg="hokuyo_node" type="hokuyo_node" name="rear_hokuyo" output="screen" respawn="true">
    <param name="port"           value="/dev/sensors/hokuyo_H0904847"/>
    <param name="frame_id"       value="rear_scanner"/>

    <param name="intensity"      value="false"/>
    <param name="calibrate_time" value="true"/>

    <param name="min_ang"        value="-2.0862"/>	
    <param name="max_ang"        value="2.0923"/>
    <param name="skip"           value="0"/>
    <param name="time_offset"    value="0.0"/>
    <param name="cluster"        value="1"/>

    <remap from="scan" to="rear_scan"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="world_to_base" output="screen" args="0 0 0 0 0 0 world base 10"/>

  <node pkg="tf" type="static_transform_publisher" name="base_to_scanner_base" output="screen" args="-0.15 0 0 0 0 0 base scanner_base 10"/>

  <node pkg="tf" type="static_transform_publisher" name="scanner_base_to_front_scanner" output="screen" args="0.025 0 0 0 0 0 scanner_base front_scanner 10"/>

  <node pkg="tf" type="static_transform_publisher" name="scanner_base_to_rear_scanner" output="screen" args="-0.025 0 0 3.14159265 0 0 scanner_base rear_scanner 10"/>

  <node pkg="obstacle_detector" type="scans_merger" name="scans_merger" output="screen">
    <param name="frame_id"          value="scanner_base"/>

    <param name="publish_scan"      value="true"/>
    <param name="publish_pcl"       value="false"/>

    <param name="ranges_num"        value="1000"/>

    <param name="max_scanner_range" value="10.0"/>
    <param name="min_scanner_range" value="0.05"/>

    <param name="max_x_range"       value="10.0"/>
    <param name="min_x_range"       value="-10.0"/>
    <param name="max_y_range"       value="10.0"/>
    <param name="min_y_range"       value="-10.0"/>
  </node>
</launch>

